rust/chains/hyperlane-cosmos/src/providers/rpc.rs

```diff
@@ -128,22 +153,26 @@ impl CosmosWasmIndexer {
     // made by the contract we are indexing.
     fn handle_tx<T>(
         &self,
-        tx: tx::Response,
+        block: BlockResponse,
+        tx_events: Events,
+        tx_hash: H256,
+        block_number: u32,
+        transaction_index: usize,
         parser: for<'a> fn(&'a Vec<EventAttribute>) -> ChainResult<ParsedEvent<T>>,
     ) -> impl Iterator<Item = (T, LogMeta)> + '_
     where
         T: PartialEq + 'static,
     {
-        tx.tx_result.events.into_iter().enumerate().filter_map(move |(log_idx, event)| {
-            if event.kind.as_str() != self.target_event_kind {
+        tx_events.events.into_iter().enumerate().filter_map(move |(log_idx, event)| {
+            if event.typ.as_str() != self.target_event_kind || !event.typ.as_str().starts_with(Self::WASM_TYPE) {
```

Comment:
why was the or condition needed?

Reply:
it was there initially and I kept it for completeness

Reply:
I'm not so sure we should keep it, iiuc it results in us trying to parse any wasm event? even though we have later checks to make sure the contract address of the event is correct, I think it's potentially unsafe - e.g. if the contract emits a couple different events with the same structure but different name, I think we may risk parsing undesired events